builtins/src/collections.rs

```diff
@@ -366,48 +463,69 @@ Convert a vector to a list.
 Section: vector
 ",
     );
+    intern_hash_haskey(env);
+
+    add_builtin(
+        env,
+        "vec-clear!",
+        vec_clear,
+        r#"Usage: (vec-clear! vector)
+
+Clear all elements from a vector. This is a destructive form!
+
+Section: vector
+
+Example:
+(def test-vec [1 2 3])
+(test::assert-false (empty? test-vec))
+(vec-clear! test-vec)
+(test::assert-true (empty? test-vec))
+(test::assert-equal [] test-vec)
+    "#,
+    );
+
+    add_builtin(
+        env,
+        "vec-remove!",
+        vec_remove,
+        r#"Usage: (vec-remove! vector index) -> vector
+
+Remove the element at index from vector, shifting all elements after it to the left.
+This is destructive!
+
+Section: vector
+
+Example:
+(def test-remove-nth-vec [1 2 3])
+(test::assert-equal [1 2 3] test-remove-nth-vec)
+(vec-remove! test-remove-nth-vec 1)
+(test::assert-equal [1 3] test-remove-nth-vec)
+(vec-remove! test-remove-nth-vec 1)
+(test::assert-equal [1] test-remove-nth-vec)
+(vec-remove! test-remove-nth-vec 0)
+(test::assert-equal [] test-remove-nth-vec)
+    "#,
+    );
 
-    /*  XXXX add these
-        add_docstring(
-            env,
-            "vec-remove!",
-            "Usage: (vec-remove! vector index) -> vector
-
-    Remove the element at index from vector, shifting all elements after it to the left.
-    This is destructive!
-
-    Section: vector
-
-    Example:
-    (def test-remove-nth-vec (vec 1 2 3))
-    (test::assert-equal '(1 2 3) test-remove-nth-vec)
-    (vec-remove! test-remove-nth-vec 1)
-    (test::assert-equal '(1 3) test-remove-nth-vec)
-    (vec-remove! test-remove-nth-vec 1)
-    (test::assert-equal '(1) test-remove-nth-vec)
-    (vec-remove! test-remove-nth-vec 0)
-    (test::assert-equal '() test-remove-nth-vec)
-    ",
-        );
-        add_docstring(
-            env,
-            "vec-insert!",
-            "Usage: (vec-insert! vector index new-element) -> vector
-
-    Inserts new-element at index and moves following elements right in vector.  This is destructive!
-
-    Section: vector
-
-    Example:
-    (def test-insert-nth-vec (vec 1 2 3))
-    (test::assert-equal '(1 2 3) test-insert-nth-vec)
-    (vec-insert! test-insert-nth-vec 1 5)
-    (test::assert-equal '(1 5 2 3) test-insert-nth-vec)
-    (vec-insert! test-insert-nth-vec 2 6)
-    (test::assert-equal '(1 5 6 2 3) test-insert-nth-vec)
-    (vec-insert! test-insert-nth-vec 0 4)
-    (test::assert-equal '(4 1 5 6 2 3) test-insert-nth-vec)
-    ",
-        );
-         */
+    add_builtin(
+        env,
+        "vec-insert!",
+        vec_insert,
+        r#"Usage: (vec-insert! vector index new-element) -> vector
+
+Inserts new-element at index and moves following elements right in vector.  This is destructive!
+
+Section: vector
+
+Example:
+(def test-insert-nth-vec [1 2 3])
```

Comment:
Just a note, this PR did not create this test.  We should start moving examples to more idiomatic (and figuring that out) slosh code.  Using def instead of a let binding for instance should be discouraged.  I did this a LOT but have been trying to clean up when working on code that intersects with docs.